tests/frontend/cli/test_starship.py

```diff
@@ -0,0 +1,506 @@
+    def test_configure_token(self, token_option):
+        """Test that specifying configure, --token and passing an argument to
+        the CLI sets the correct attribute."""
+        parser = cli.create_parser()
+
+        args = parser.parse_args(['configure', token_option, 'SomeToken'])
+        assert args.func is cli.configure
+        assert args.token == 'SomeToken'
+        assert not args.local
+
+    def test_configure_everything(self):
+        """Test that specifying configure, --local to the CLI sets the correct
+        attribute."""
+        parser = cli.create_parser()
+        args = parser.parse_args(['configure'])
+
+        assert args.func is cli.configure
+        assert not args.local
+
+    @pytest.mark.parametrize("token_option", ['--token', '-t'])
+    @pytest.mark.parametrize("local_option", ['--local', '-l'])
+    def test_configure_token_locally(self, token_option, local_option):
+        """Test that specifying configure, --token, --local and passing an argument to
+        the CLI sets the correct attribute."""
+        parser = cli.create_parser()
+
+        args = parser.parse_args(['configure', token_option, 'SomeToken', local_option])
+        assert args.func is cli.configure
+        assert args.token == 'SomeToken'
+
+    @pytest.mark.parametrize("option", ['--local', '-l'])
+    def test_configure_everything_locally(self, option):
+        """Test that specifying configure, --local to the CLI sets the correct
+        attribute."""
+        parser = cli.create_parser()
+        args = parser.parse_args(['configure', option])
+
+        assert args.func is cli.configure
+        assert args.local
+
+    def test_run(self):
+        """Test that specifying input and passing an argument to the CLI sets
+        the correct attribute."""
+        parser = cli.create_parser()
+        args = parser.parse_args(['run', 'SomePath'])
+
+        assert args.func is cli.run_blackbird_script
+        assert args.input == 'SomePath'
+
+    def test_output(self):
+        """Test that specifying input, --output and passing the arguments to
+        the CLI sets the correct attributes."""
+        parser = cli.create_parser()
+        args = parser.parse_args(['run', 'SomeInputPath', '--output', 'SomeOutputPath'])
+
+        assert args.func is cli.run_blackbird_script
+        assert args.input == 'SomeInputPath'
+        assert args.output == 'SomeOutputPath'
+
+class MockArgs:
+    """A mock class used for mocking the args that are parsed from the command
+    line."""
+
+    def __init__(self):
+        self.token = None
+        self.local = None
+        self.input = None
+        self.output = None
+
+class MockStoreAccount:
+    """A mock class used for capturing the arguments with which the store_account
+    function is being called."""
+
+    def __init__(self):
+        self.kwargs = None
+
+    def store_account(self, **kwargs):
+        self.kwargs = kwargs
```

Comment:
How come some of the mocking functions/classes are here, and some are in conftest? Not suggesting any change here, just curious!

Reply:
See my comment on this above (namely the `conftest` was eventually not used, but can move any of these mocked functions&classes there if it would be better that way).